src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAccountManager.java

```diff
@@ -1339,16 +1344,26 @@ public void mergeUserAccounts(final RegisteredUserDTO target, final RegisteredUs
      * @throws NoUserException
      *             - If no user found with provided email.
      */
-    public final void resetPasswordRequest(final RegisteredUserDTO userObject) throws InvalidKeySpecException,
-            NoSuchAlgorithmException, CommunicationException, SegueDatabaseException, NoUserException {
+    public final boolean resetPasswordRequest(final RegisteredUserDTO userObject)
+            throws SegueDatabaseException {
         RegisteredUser user = this.findUserByEmail(userObject.getEmail());
 
         if (null == user) {
-            throw new NoUserException("No user found with this email!");
+            return false;
         }
 
-        RegisteredUserDTO userDTO = this.convertUserDOToUserDTO(user);
-        this.userAuthenticationManager.resetPasswordRequest(user, userDTO);
+        executor.submit(() -> {
```

Comment:
what's the reason behind threading this?

Reply:
As per the ticket, we want this function is execute in approximately the same time and provide an identical response regardless of whether the specified user account exists.
Pretty much everything past that point requires a valid user object but doesn't return anything critical to the response. As such, it should be fine to hand off to an asynchronous process here and return a response?

Reply:
yep, I think I figured this out after writing the note and forgot to come back and delete it. This is fine, a nice clean way of achieving it.